### PR TITLE
add Levinstein distance with Dynamic Programming: up -> down approach

### PR DESCRIPTION
**Pull request by alexpantyukhin**
### Describe your change:
Levinstein distance with Dynamic Programming: up -> down approach. some tests are copied from https://github.com/TheAlgorithms/Python/blob/master/dynamic_programming/edit_distance.py#L61 .

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by alexpantyukhin** [comment]
The task: #7171

File: dynamic_programming/min_distance_up_bottom.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    10
+    """
+
+    from functools import lru_cache
+
+    len_word1 = len(word1)
+    len_word2 = len(word2)
+
+    @lru_cache(maxsize=None)
+    def min_distance(index1, index2):
```
[the remark this answers is not in the document]
Comment (alexpantyukhin) [reply dynamic_programming/min_distance_up_bottom.py]: fixed

File: dynamic_programming/min_distance_up_bottom.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    10
+    """
+
+    from functools import lru_cache
+
+    len_word1 = len(word1)
+    len_word2 = len(word2)
+
+    @lru_cache(maxsize=None)
+    def min_distance(index1: int, index2: int) -> int:
```
[the remark this answers is not in the document]
Comment (alexpantyukhin) [reply dynamic_programming/min_distance_up_bottom.py:35]: this is an inner function. doc tests cover this function

**Comment by alexpantyukhin** [comment]
There is an inner function in the implementation. It doesn't requires tests, because outer function is tested.

**Comment by alexpantyukhin** [comment]
@Kush1101 could you please review this PR

**Review by CaedenPH — APPROVED** [review]
@cclauss Looks good

File: dynamic_programming/min_distance_up_bottom.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        # if second word index is overflow - delete all from the first word
+        if index2 >= len_word2:
+            return len_word1 - index1
+
+        # if letters on the current positions are identical - no diff.
+        diff = 0
+        if word1[index1] != word2[index2]:
+            diff = 1
+
```
Comment (cclauss) [inline dynamic_programming/min_distance_up_bottom.py:49]:
```suggestion
        diff = int(word1[index1] != word2[index2])  # diff is current letters are not identical
```

**Review by cclauss — APPROVED** [review]

**Pull request merged**